simple_app/app_simple.py

```diff
@@ -171,6 +187,141 @@ def autocomplete():
     return jsonify(alljsonlist=all_results)  # wraps all of the object names as a list, into a .json for server use
 
 
+@app_simple.route('/write/<key>', methods=['GET', 'POST'])
+def create_file_for_download(key: str):
+    """
+    Creates and downloads the shown dataframe from solo results
+
+    Parameters
+    ----------
+    key: str
+        The dataframe string
+    """
+    query = curdoc().template_variables['query']
+    db = SimpleDB(db_file, connection_arguments={'check_same_thread': False})  # open database
+    resultdict: dict = db.inventory(query)  # get everything about that object
+    everything = Inventory(resultdict, args, rtnmk=False)
+    if key in resultdict:
+        results: pd.DataFrame = getattr(everything, key.lower())
+        fname = write_file(results).split('/')[-1]
+        return redirect(url_for('download_file', filename=fname))
+    return None
+
+
+@app_simple.route('/write_spectra', methods=['GET', 'POST'])
+def create_spectrafile_for_download():
+    """
+    Downloads the spectra files and zips together
```

Comment:
At MAST, this approach has caused issues from time to time. The logic being that you are downloading from the cloud to a local disk, then bundling it and sending it back. It is far better to just point to use to the individual files and have them get it from the source. Which means no bundling of files, or at best, generating a script for them to download on their end.